controllers/discovery/initDiscoveryUserData.js

```diff
@@ -69,11 +88,21 @@ const InitDiscoveryUserData = async (req, res) => {
     });
     let suggestedUsers = usersWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData[0]?.total_data;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedUsers,
-      nextPage: page + 1
+      nextPage: page + 1,
+      total_page: Math.ceil(totalData / limit)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is no error handling for the case when `totalData` is undefined or not a number. This could lead to unexpected behavior or errors. Consider adding a check for this.

```diff
- totalData = totalData[0]?.total_data;
+ totalData = totalData && totalData[0] ? totalData[0].total_data : 0;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->